### ISSUE TITLE
HashMap.equals() is broken

### ISSUE THREAD

**Issue by orionll** [javaslang/javaslang#1815]
```java
HashMap<String, Integer> map = HashMap.<String, Integer> empty().put("Aa", 1).put("BB", 2);
HashMap<String, Integer> map2 = HashMap.<String, Integer> empty().put("BB", 2).put("Aa", 1);
System.out.println(map.equals(map2)); // prints false
```

`map` and `map2` contain the same key-value pairs, but `equals()` returns `false`.

**Comment by ruslansennov** [comment]
@orionll nice catch, thank you

**Comment by orionll** [comment]
@ruslansennov There is one more bug: `map.hashCode() != map2.hashCode()`. Two equal objects must have the same hash codes. This is very easy to fix: just return the sum of hash codes of each key-value pair.

**Comment by orionll** [comment]
And HashSet.hashCode() is also incorrect.

**Comment by ruslansennov** [comment]
Sh*t...
You're right, thank you

**Comment by ruslansennov** [comment]
fixed #1823
